bundles/org.openhab.io.homekit/src/main/java/org/openhab/io/homekit/internal/HomekitOHItemProxy.java

```diff
@@ -0,0 +1,134 @@
+ *
+ */
+
+public class HomekitOHItemProxy {
+    private static final Logger logger = LoggerFactory.getLogger(HomekitOHItemProxy.class);
+    public static final String HUE_COMMAND = "hue";
+    public static final String SATURATION_COMMAND = "saturation";
+    public static final String BRIGHTNESS_COMMAND = "brightness";
+    public static final String ON_COMMAND = "on";
+
+    // delay, how long wait for further commands.
+    // TODO: make it configurable ?
+    private final int DELAY = 50;
+
+    private final ScheduledExecutorService scheduler = ThreadPoolManager
+            .getScheduledPool(ThreadPoolManager.THREAD_POOL_NAME_COMMON);
+    private ScheduledFuture<?> future;
+
+    private final Item item;
+
+    @Nullable
+    private ConcurrentHashMap<String, State> commandCache;
+
+    public HomekitOHItemProxy(final Item item) {
+        this.item = item;
+    }
+
+    public Item getItem() {
+        return item;
+    }
+
+    private void sendCommand() {
+        if (item instanceof ColorItem) {
+            final State currentState = item.getState() instanceof UnDefType ? HSBType.BLACK : item.getState();
+            final State hue = commandCache.containsKey(HUE_COMMAND) ? commandCache.remove(HUE_COMMAND)
+                    : ((HSBType) currentState).getHue();
+            final State saturation = commandCache.containsKey(SATURATION_COMMAND)
+                    ? commandCache.remove(SATURATION_COMMAND)
+                    : ((HSBType) currentState).getSaturation();
+            final State brightness = commandCache.containsKey(BRIGHTNESS_COMMAND)
+                    ? commandCache.remove(BRIGHTNESS_COMMAND)
+                    : ((HSBType) currentState).getBrightness();
+            ((ColorItem) item).send(new HSBType((DecimalType) hue, (PercentType) saturation, (PercentType) brightness));
+            logger.trace("send HSB command for item {} with following values h {} s{} b{}", item, hue, saturation,
+                    brightness);
+        } else if (item instanceof DimmerItem) { // not ColorItem but DimmerItem
+            final State on = commandCache.remove(ON_COMMAND);
+            final State brightness = commandCache.remove(BRIGHTNESS_COMMAND);
+
+            // if "ON" command received we dont need to send brightness.
+            // TODO: make brightness suppression on "On" configurable.
+            if (on != null) {
+                logger.trace("send OnOff command for item {} with value {}", item, on);
+                ((DimmerItem) item).send((OnOffType) on);
+            } else {
+                if (brightness != null) {
+                    logger.trace("send brightness command for item {} with value {}", item, brightness);
+                    ((DimmerItem) item).send((PercentType) brightness);
+                }
+            }
+        }
+    }
+
+    public synchronized void sendCommandProxy(final String commandType, final State state) {
+        if (commandCache == null) {
```

Comment:
For null safety I would prefer to instantiate on object creation I don't think we'll have so many items that an empty HashMap creates a significant burden on the whole system

Reply:
my reasoning was -  i have around 80 items and 0 dimmer and 0 hue devices, so, it will create 80 empty hashMap without having usage for it. 
hm.. but actually, on other side, it will also create 80 HomekitOHItemProxy without real need for it. 
so, if we go for optimisation to probably at much earlier point - dont create HomekitOHItemProxy in first place which makes the logic even more complex.